### PR TITLE
Run tests for all platforms in dev build

### PR DESCRIPTION
**Pull request by andrei-epure-sonarsource**
This should have been done in #3700 ... better late than never.

In #3700 I forced the same behavior as on AZP.

By running `.\scripts\build\dev-build.ps1 -build -test -release` before opening a PR we'll make sure we won't have surprises.

### CODE REVIEW

File: scripts/build/build-utils.ps1 [hunk 1]
```diff
@@ -147,6 +147,15 @@ function Invoke-UnitTests([string]$binPath, [bool]$failsIfNotTest) {
 
     & (Get-VsTestPath) $testFiles /Parallel /Enablecodecoverage /InIsolation /Logger:trx /TestAdapterPath:$testDirs
     Test-ExitCode "ERROR: Unit Tests execution FAILED."
+
```
Comment (pavel-mikula-sonarsource) [inline scripts/build/build-utils.ps1:150]: I might be useful to add log like `Write-Header "Running unit tests .NET Core 3.1"` to report the stage. Also update the message for the first run on line 131

File: scripts/build/build-utils.ps1 [hunk 2]
```diff
@@ -147,6 +147,15 @@ function Invoke-UnitTests([string]$binPath, [bool]$failsIfNotTest) {
...
+    $testProjFileName = "tests\SonarAnalyzer.UnitTest\SonarAnalyzer.UnitTest.csproj"
+    & dotnet build $testProjFileName
```
Comment (pavel-mikula-sonarsource) [inline scripts/build/build-utils.ps1:152]: Line 138 expects `*.UnitTest.dll` to exist already. Why is it build here then?

**Review by pavel-mikula-sonarsource — CHANGES_REQUESTED** [review]
Improvement suggestion and one question that confuses me.

File: scripts/build/build-utils.ps1 (hunk 2 above)
Comment (andrei-epure-sonarsource) [reply scripts/build/build-utils.ps1]: I actually had to pass the buildConfiguration given to the script

**Review by pavel-mikula-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**